### PR TITLE
feat() add gcp dev pulumi configuration files

### PR DESCRIPTION
**Pull request by bartsmykla**
Configuration files to setup GCP/GKE clusters for kuma to test/development purposes

### CODE REVIEW

File: gcp-devel/go.mod [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module gcp-devel
+
+go 1.14
```
Comment (jpeach) [inline gcp-devel/go.mod]: Nit: can we bump this to 1.16?

File: gcp-devel/main.go [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+	NodeConfig    ConfigClustersNodeConfig
+	Names         []string
+}
+
+type Config struct {
+	Clusters ConfigClusters
+	Location string
+}
+
+// GenerateSSHKeys ...
```
Comment (jpeach) [inline gcp-devel/main.go:52]: Nit: Add a TODO to collapse this in to the keygen code from aws-devel

File: gcp-devel/main.go [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+	// }
+	//
+	// defer f.Close()
+
+	return privBuffer.String(), base64.StdEncoding.EncodeToString(public.Marshal()), nil
+}
+
+var (
+	PrivateKeySecretName = "kuma-main-ssh-private-key"
+	PublicKeySecretName  = "kuma-main-ssh-public-key"
```
Comment (jpeach) [inline gcp-devel/main.go:84]: yeh we should come back and standardize this with `aws-devel`

File: gcp-devel/main.go [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+			createPrivKey := fmt.Sprintf("printf -- '%s' | gcloud secrets create %s --data-file=-", privateKey, PrivateKeySecretName)
+			createPubKey := fmt.Sprintf("printf -- '%s' | gcloud secrets create %s --data-file=-", publicKey, PublicKeySecretName)
+
+			_ = ctx.Log.Info(createPrivKey, nil)
+			_ = ctx.Log.Info("###", nil)
+			_ = ctx.Log.Info(createPubKey, nil)
+
+			return errors.New("cannot proceed without necessary ssh keys")
+		} else {
+			publicKeySecret, err := secretmanager.LookupSecretVersion(ctx, &secretmanager.LookupSecretVersionArgs{
```
Comment (jpeach) [inline gcp-devel/main.go]: Nit drop the else and outdent its contents.

**Review by jpeach — APPROVED** [review]

File: gcp-devel/go.mod (hunk 1 above)
Comment (bartsmykla) [reply gcp-devel/go.mod]: Addressed

File: gcp-devel/main.go (hunk 2 above)
Comment (bartsmykla) [reply gcp-devel/main.go:52]: Addressed

File: gcp-devel/main.go (hunk 4 above)
Comment (bartsmykla) [reply gcp-devel/main.go]: Addressed

**Comment by bartsmykla** [comment]
Hi James, I addressed most of your comments. Could you merge the PR and we will followup with the standarization of key names in separate PR

**Review by jpeach — APPROVED** [review]
LGTM

**Pull request merged**